src/workers/parser/steps.js

```diff
@@ -1,30 +1,30 @@
 import { Source } from "../../../flow-typed/debugger-html";
 import { AstPosition } from "./types";
 import { getClosestPath } from "./utils/closest";
-import { isAwaitExpression } from "./utils/helpers";
+import { isAwaitExpression, isYieldExpression } from "./utils/helpers";
 import type { NodePath } from "babel-traverse";
 
 export function getNextStep(source: Source, pausedPosition: AstPosition) {
-  const awaitExpression = getAwaitExpression(source, pausedPosition);
+  const awaitExpression = getAsyncExpression(source, pausedPosition);
   if (!awaitExpression) {
     return null;
   }
   const awaitStatement = awaitExpression.getStatementParent();
   return getLocationAfterAwaitExpression(awaitStatement, pausedPosition);
 }
 
-function getAwaitExpression(source: Source, pausedPosition: AstPosition) {
+function getAsyncExpression(source: Source, pausedPosition: AstPosition) {
   const closestPath = getClosestPath(source, pausedPosition);
 
   if (!closestPath) {
     return null;
   }
 
-  if (isAwaitExpression(closestPath)) {
+  if (isAwaitExpression(closestPath) || isYieldExpression(closestPath)) {
```

Comment:
this should be separated, or the function renamed, since yield is not async and this would lead to misunderstandings later on.

Reply:
I ended up going with `getSteppableExpression` for now as separating out the two cases led to more conditional logic living in `getNextStep`.